src/consensus.rs

```diff
@@ -826,75 +710,34 @@ impl Consensus {
                 "Slot {} PrepareVote message validated. Broadcasting Confirm",
                 self.bft_round_state.slot
             );
-            self.bft_round_state.step = Step::ConfirmAck;
             self.broadcast_net_message(ConsensusNetMessage::Confirm(
                 consensus_proposal_hash.clone(),
-                self.bft_round_state.prepare_quorum_certificate.clone(),
+                QuorumCertificate {
+                    signature: prepvote_signed_aggregation.signature,
+                    validators: prepvote_signed_aggregation.validators,
+                },
             ))?;
         }
         // TODO(?): Update behaviour when having more ?
         // else if validated_votes > 2 * f + 1 {}
         Ok(())
     }
 
-    fn verify_prepare_qc(&self, 
-        consensus_proposal_hash: &ConsensusProposalHash,
-        prepare_quorum_certificate: &QuorumCertificate,
-    ) -> Result<()> {
-        // Verifies the *Prepare* Quorum Certificate
-        let prepare_quorum_certificate_with_message = SignedWithKey {
-            msg: ConsensusNetMessage::PrepareVote(consensus_proposal_hash.clone()),
-            signature: prepare_quorum_certificate.signature.clone(),
-            validators: prepare_quorum_certificate.validators.clone(),
-        };
-
-        match BlstCrypto::verify(&prepare_quorum_certificate_with_message) {
-            Ok(res) if !res => {
-                self.metrics.confirm_error("prepare_qc_invalid");
-                bail!("Prepare Quorum Certificate received is invalid")
-            }
-            Ok(_) => Ok(()),
-            Err(err) => bail!("Prepare Quorum Certificate verification failed: {}", err),
-        }                    
-    }
-
     /// Message received by follower.
     fn on_confirm(
```

Comment:
Pas besoin de check de Step pour le on_confirm ?

Reply:
Je me demande si l'utilisation du 
 `if voting_power > 2 * f {` au lieu de
 ` if voting_power == 2 * f +1 {` dans on_prepare_vote couplé au non-check de Step ici, fait qu'on va reprocess la request, donc et resend un confirmAck
 C'est ptretre pas la mort in the end :shrug:

Reply:
Pas besoin de check le step -> c'est que pour le leader (pour le moment)